hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupInfoManagerImpl.java

```diff
@@ -1181,4 +1224,9 @@ public void moveServers(Set<Address> servers, String targetGroupName) throws IOE
     }
   }
 
+  @Override
+  public String determineRSGroupInfoForTable(TableName tableName) {
+    return script.getRSGroup(tableName.getNamespaceAsString(), tableName.getQualifierAsString());
```

Comment:
What about store the group info as a TableDescriptor attribute?

Reply:
> What about store the group info as a TableDescriptor attribute?

Master branch have already done so.

This feature is mainly about when client specified TableDescriptor doesn't have rsgroup configured, and admin want to intercept it before table assignments. It's from server side perspective.

Reply:
Ok, get it.